src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -105,15 +105,15 @@ select k
 
             PropertyNameToCanonicalKeyword = properties.ToDictionary(
                 p => p.Name,
-                p => p.GetCustomAttribute<DisplayNameAttribute>().DisplayName
+                p => p.GetCustomAttribute<DisplayNameAttribute>()!.DisplayName
             );
 
             PropertyDefaults = properties
                 .Where(p => p.GetCustomAttribute<ObsoleteAttribute>() == null)
                 .ToDictionary(
                 p => p,
                 p => p.GetCustomAttribute<DefaultValueAttribute>() != null
-                    ? p.GetCustomAttribute<DefaultValueAttribute>().Value
+                    ? p.GetCustomAttribute<DefaultValueAttribute>()!.Value
```

Comment:
Can capture the attribute above in the check (with pattern matching) and then you don't need to repeat the search or the bang.

Reply:
Opened #2564. Will remove this magic.

Reply:
I'm not sure we need to remove this (see comment in #2564). In any case, in the meantime let's clean up the bangs...